### ISSUE TITLE
Buffering behaviour of Web API under Python 3.8

### ISSUE THREAD

**Issue by Morendil** [openfisca/openfisca-core#950]
Yesterday while debugging with @sandcha I noticed that the Web API is emitting warnings related to buffering under Python 3.8, and fails to return a response immediately to a request via `curl` or via web browser.

The response eventually appears on the client after killing the server process, which suggests the output is being buffered but never sent.

Additionally, it felt subjectively as if the Web API (`openfisca serve`) was particularly slow to start even after going back to Python 3.7 but we didn't benchmark this.

**Comment by maukoquiroga** [comment]
Thanks @Morendil ! However I'm not able to reproduce the issue :

Terminal 1:

```
hyperion@diotimac:~/Sites/dinsic/openfisca/openfisca-core[fix-buffering] > python --version
Python 3.8.0
hyperion@diotimac:~/Sites/dinsic/openfisca/openfisca-core[fix-buffering] > make api
openfisca serve --country-package openfisca_country_template --extensions openfisca_extension_template
[2020-02-22 21:00:37 +0100] [13139] [INFO] Starting gunicorn 20.0.4
[2020-02-22 21:00:37 +0100] [13139] [INFO] Listening at: http://127.0.0.1:5000 (13139)
[2020-02-22 21:00:37 +0100] [13139] [INFO] Using worker: sync
[2020-02-22 21:00:37 +0100] [13171] [INFO] Booting worker with pid: 13171
[2020-02-22 21:00:37 +0100] [13172] [INFO] Booting worker with pid: 13172
[2020-02-22 21:00:37 +0100] [13173] [INFO] Booting worker with pid: 13173
[2020-02-22 21:00:59 +0100] [13139] [INFO] Handling signal: winch
[2020-02-22 21:00:59 +0100] [13139] [INFO] Handling signal: winch
[2020-02-22 21:00:59 +0100] [13139] [INFO] Handling signal: winch
```

Terminal 2:

```
hyperion@diotimac:~/Sites/dinsic/openfisca/openfisca-core[fix-buffering] > python --version
Python 3.8.0
hyperion@diotimac:~/Sites/dinsic/openfisca/openfisca-core[fix-buffering] > curl http://127.0.0.1:5000/
*   Trying 127.0.0.1:5000...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 5000 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:5000
> User-Agent: Mozilla/5.0 (Windows NT 6.1; rv:31.0) Gecko/20100101 Firefox/31.0
> Accept: */*
> Referer: 
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 300 MULTIPLE CHOICES
< Server: gunicorn/20.0.4
< Date: Sat, 22 Feb 2020 20:01:47 GMT
< Connection: close
< Content-Type: application/json
< Content-Length: 219
< Country-Package: openfisca-country-template
< Country-Package-Version: 3.9.10
< Access-Control-Allow-Origin: *
< 
{"welcome":"This is the root of an OpenFisca Web API. To learn how to use it, check the general documentation (https://openfisca.org/doc/) and the OpenAPI specification of this instance (http://127.0.0.1:5000/spec)."}
* Closing connection 0
```

Result of `pip list`:

```
Package                      Version    Location                                             
---------------------------- ---------- -----------------------------------------------------
attrs                        19.3.0     
autopep8                     1.5        
bleach                       3.1.1      
certifi                      2019.11.28 
chardet                      3.0.4      
Click                        7.0        
coverage                     5.0.3      
docutils                     0.16       
dpath                        1.5.0      
entrypoints                  0.3        
flake8                       3.7.9      
flake8-bugbear               19.8.0     
flake8-print                 3.1.4      
Flask                        1.1.1      
Flask-Cors                   3.0.7      
gunicorn                     20.0.4     
idna                         2.9        
itsdangerous                 1.1.0      
Jinja2                       2.11.1     
keyring                      21.1.0     
MarkupSafe                   1.1.1      
mccabe                       0.6.1      
mock                         4.0.1      
more-itertools               8.2.0      
mypy                         0.761      
mypy-extensions              0.4.3      
numexpr                      2.7.1      
numpy                        1.17.5     
OpenFisca-Core               34.7.2     /Users/hyperion/Sites/dinsic/openfisca/openfisca-core
OpenFisca-Country-Template   3.9.10     
OpenFisca-Extension-Template 1.3.3      
packaging                    20.1       
pip                          20.0.2     
pkginfo                      1.5.0.1    
pluggy                       0.13.1     
psutil                       5.7.0      
py                           1.8.1      
pycodestyle                  2.5.0      
pyflakes                     2.1.1      
Pygments                     2.5.2      
pyparsing                    2.4.6      
pytest                       5.3.5      
pytest-cov                   2.8.1      
pytest-mock                  2.0.0      
PyYAML                       5.3        
readme-renderer              24.0       
requests                     2.23.0     
requests-toolbelt            0.9.1      
setuptools                   41.2.0     
six                          1.14.0     
sortedcontainers             2.1.0      
tqdm                         4.43.0     
twine                        3.1.1      
typed-ast                    1.4.1      
typing-extensions            3.7.4.1    
urllib3                      1.25.8     
wcwidth                      0.1.8      
webencodings                 0.5.1      
Werkzeug                     1.0.0      
wheel                        0.34.2     
```

**Comment by Morendil** [comment]
Might have been seeing (for some reason) pre-20.0.x gunicorn behaviour based on [this issue](https://github.com/benoitc/gunicorn/issues/2091). Or some other call in a dependency since updated.
